balrog/os-de-support.rst

```diff
@@ -0,0 +1,198 @@
+
+* Create a new De-Support release referencing the SUMO KB article. Careful: Use a locale-agnostic link like https://support.mozilla.org/kb/firefox-users-macos-1012-1013-1014-moving-to-extended-support (instead of https://support.mozilla.org/en-US/kb/firefox-users-macos-1012-1013-1014-moving-to-extended-support). Example release blob: ::
+
+    {
+        "detailsUrl": "https://support.mozilla.org/kb/firefox-users-macos-1012-1013-1014-moving-to-extended-support",
+        "displayVersion": "115.0",
+        "name": "OSX-10.12-10.14-Desupport",
+        "product": "Firefox",
+        "schema_version": 50
+    }
+
+* Create a rule to serve the de-support notice: match the ``Firefox : Nightly*`` channel and the affected OS version with lower priority than the pinning rule (so probably those who have been updated to the last supported build); serve the new De-Support release.
+* Delete the ``NoUpdate`` rule created earlier.
+
+Update Balrog rules for Beta and Aurora
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Check with the *Product* organization to verify per-channel requirements for Beta and DevEdition.
+
+To implement DE-SUPPORT for DevEdition, follow the de-support procedure for Nightly, above, with these changes:
+
+* Make changes just before and after the Merge Day II merge of central to beta.
+* Use the ``Firefox : aurora*`` channel
+* There should be no need to add aliases to the release blob.
+
+To implement ESR-SWITCH for Beta, follow the de-support procedure for Release, below, with these changes:
+
+* Make changes just before and after the Merge Day II merge of central to beta.
+* Use the ``Firefox : beta*`` channel
+* There should be no need to add aliases to the release blob.
+
+Update Balrog rules for Release
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For the Release channel, we generally ESR-SWITCH: serve an update referencing the channel-switching MAR, so that
+users on the affected OS move from the Release channel to the ESR channel.
+
+Just *before* the first release build of the first unsupported version, pause Release updates for the affected OS version. This will typically be just before the Merge Day I merge of Beta to Release, for the release after the Beta changes.
+
+* Create a new rule matching the ``Firefox : Release*`` channel and the OS version to serve the ``NoUpdate`` release.
+
+Shortly *after* the first release build of the first unsupported version:
+
+* Create a new release to serve the channel-switching mar. *TBD: do we duplicate an existing release, then manually replace the url?*
+* Create a new rule matching the ``Firefox : Release*`` channel and the OS version to serve the channel-switching release.
+* *TBD: Is that all, or do we need pinning?*
+* Delete the ``NoUpdate`` rule created earlier.
+ 
+Update Balrog rules for ESR
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TBD
+
+* Like https://bugzilla.mozilla.org/show_bug.cgi?id=1275609 ?
+* Pin and de-support?
+* How do we remember to do this? File a bug now and .... ?
```

Comment:
When an esr branch goes EOL we should go through the release channel rules between that esr major version and the next, to figure out whether we need watersheds etc, and the no-longer-supported OS versions should come up at that point.

Reply:
Thanks, that makes sense. In that case, it sounds like that process will usually just work, and I don't see any point in describing it here. I'll keep this section brief, at least for now.